### PR TITLE
Authentication using Sanctum

### PR DESCRIPTION
**Pull request by Bahugunajii**
## Description
- Implemented authentication using sanctum

## Checklist
- [x] Self review

### CODE REVIEW

File: database/migrations/2023_03_06_174954_add_device_name_to_users_table.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('device_name');
+        });
+    }
+};
```
Comment (Ayush8923) [inline database/migrations/2023_03_06_174954_add_device_name_to_users_table.php]: What are you using this table for? I think we don't need this table.

File: app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+use Illuminate\Validation\ValidationException;
+
+class AuthController extends Controller
+{
+    public function login(Request $request) {
+        $request->validate([
+            'email' => 'required|email',
+            'password' => 'required',
+            'device_name' => 'required',
+        ]);
```
Comment (Ayush8923) [inline app/Http/Controllers/AuthController.php]: I think we need to create a new file for the `request` Validator.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]

File: database/migrations/2023_03_06_174954_add_device_name_to_users_table.php (hunk 1 above)
Comment (Bahugunajii) [reply database/migrations/2023_03_06_174954_add_device_name_to_users_table.php]: sure... It was required only for testing purpose

File: app/Http/Controllers/AuthController.php (hunk 2 above)
Comment (Bahugunajii) [reply app/Http/Controllers/AuthController.php]: for what purpose?

Comment (Ayush8923) [reply app/Http/Controllers/AuthController.php]: To validate whether this field is filled or not. And a little code to keep its cleanup. I think we need to use the `Single Responsibility Principle` here.

File: database/seeders/UserSeeder.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{
+    /**
+     * Run the database seeds.
+     */
+    public function run(): void
+    {
+        DB::table('users')->insert([
+            'name' => 'Aman',
+            'email' => 'amanbahuguna009@gmail.com',
+            'password' => Hash::make('testpass')
```
Comment (Ayush8923) [inline database/seeders/UserSeeder.php:20]:
```suggestion
            'password' => Hash::make('password')
```

File: routes/api.php [hunk 4]
```diff
@@ -17,3 +18,6 @@
 Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
     return $request->user();
 });
+
+Route::post('/login', [AuthController::class, 'login']);
+Route::post('/register', [AuthController::class, 'register']);
```
Comment (Ayush8923) [inline routes/api.php:23]:
```suggestion
Route::post('/register', [AuthController::class, 'register']);

```

File: database/seeders/UserSeeder.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+class UserSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     */
+    public function run(): void
+    {
+        DB::table('users')->insert([
+            'name' => 'Aman',
+            'email' => 'amanbahuguna009@gmail.com',
```
Comment (Ayush8923) [inline database/seeders/UserSeeder.php:19]: Can we write like this?
```suggestion
            'name' => 'poolYourRide Admin',
            'email' => 'pollyourride@gmail.com',
```

File: app/Http/Requests/AuthRequest.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                    'email' => 'required|email',
+                    'password' => 'required',
+                    'device_name' => 'required',
+                ];
+            case 'register':
+                return [
+                    'name' => 'required|string|max:255',
+                    'email' => 'required|string|email|max:255|unique:users',
+                    'password' => 'required|string|min:8',
+                    'device_name' => 'required',
```
Comment (Ayush8923) [inline app/Http/Requests/AuthRequest.php:36]: I think we don't need to make it required field. let me know your thoughts

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left soms comments. Let me know if there are any questions

File: database/seeders/UserSeeder.php (hunk 5 above)
Comment (Bahugunajii) [reply database/seeders/UserSeeder.php:19]: It's just a seeder. Credentials can have any value

File: app/Http/Requests/AuthRequest.php (hunk 6 above)
Comment (Bahugunajii) [reply app/Http/Requests/AuthRequest.php:36]: These fields are the required fields. Refer Laravel Sanctum docs: https://laravel.com/docs/10.x/sanctum#mobile-application-authentication

File: database/seeders/UserSeeder.php (hunk 3 above)
Comment (Bahugunajii) [reply database/seeders/UserSeeder.php:20]: Password can be anything.

**Review by Ayush8923 — APPROVED** [review]

**Pull request merged**